### PR TITLE
[7534] add tito widget block

### LINKED ISSUE #7534: BE: Mozfest Tito Ticket purchase embed
Epic: #7433 

### Describe the feature request
A clear and concise description of your feature request

Figma for quick reference: https://www.figma.com/proto/2knnwWmmqCXuMyF6WPiOuG/MozFest-Additional-Features?page-id=0%3A1&node-id=23%3A37&viewport=288%2C48%2C0.27&scaling=min-zoom&starting-point-node-id=23%3A37&show-proto-sidebar=1

### Implementation Checklist
- [ ] Editor can add a new StreamField block that has the Tito widget
- [ ] Editor can specify the button label
- [ ] Editor can specify the Tito event 
- [ ] Editor can specify the Tito release(s)
Ideally checklist also includes:
- Want to have it set a redirect page so that when the user completes the purchase they are sent to a thank you page.

### PR DESCRIPTION
**Pull request by richbrennan**
Closes #7534
Related PRs/issues #7433

**Link to sample test page**: 
https://foundation-s-feature-75-9dly9w.herokuapp.com/cms/pages/163/edit/
https://mozfest-foundation-s-feature-75-9dly9w.mofostaging.net/en/tickets/

The Tito widget can be tested with the following credit card details:
Number: 4242 4242 4242 4242
Expiry: 06/22
CVC: 123
ZIP: 32003

There are a couple of outstanding issues that a FE dev might be able to help with:

1. The top of the widget is cut off due to the sticky nav at the top of the page (see screenshot below)
2. When a redirect page is added it redirects the user as soon as the Tito order complete info is displayed in the widget, so they don't get the opportunity to view it. Someone with more JS knowledge might be able to come up with a way to redirect once the widget closes, but there isn't an official Tito callback that I can hook into to achieve this: https://ti.to/docs/api/widget#tito-widget-v2-callbacks

Screenshot from 2021-11-25 17-30-56

### CODE REVIEW

**Comment by richbrennan** [comment]
I've made some changes to the app.json file that I believe review apps use but they don't seem to be being picked up. I've updated some CSP_* settings to load external scripts and styles but they're still being blocked on the review app, after working locally: https://github.com/mozilla/foundation.mozilla.org/pull/7938/files#diff-96677aa35debfefd031d9d34d9c70369754ee3acb2d9a9d4090e98612efee6f5

Screenshot from 2021-11-26 14-11-09

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/tito_widget_block.html
Comment (richbrennan) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/tito_widget_block.html:6]: Use of redirect page to-be-confirmed

Comment (Pomax) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/tito_widget_block.html:6]: If there's follow-up investigation to be done, please remember to file an issue before this PR lands.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/tito_widget_block.html
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/tito_widget_block.html:20]: Rather than inlining this, let's instead make this part of the mozfest JS bundle (entry point: `source/js/foundation/pages/mozfest/index.js`), and then with that `{{ value.redirect_page.full_url }}` as content in a queryable element (e.g. `<input type="hidden" name="tito-post-registration-url" value="{{ value.redirect_page.full_url }}">`).

So have this in a file like `.../mozfest/tito.js`

```javascript
export function setupTito() {
  globalThis.tito = globalThis.tito || function() {
    (tito.q = tito.q || []).push(arguments);
  };

  tito('on:registration:finished', function(data) {
    const redirectURL = document.querySelector(`input[name="tito-post-registration-url"]`)?.value;
    if (redirectURL) {
      globalThis.location = redirectURL;
    }
  });
};
```
and then import and run that in the `index.js` file:

```javascript
import { bindEventHandlers } from "./template-js-handler";
import { setupTito } from "./tito.js";

bindEventHandlers();
setupTito();
```

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/tito_widget_block.html
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/tito_widget_block.html:13]: What is this line supposed to do? Because if `window.tito` is undefined, any attempt to access `tito.q` in global scope is going to throw a `ReferenceError: tito is not defined` error.

**Review by Pomax — CHANGES_REQUESTED** [review]

**Comment by Pomax** [comment]
@richbrennan I see this got marked as draft, rather than being marked for rereview - was that supposed to happen?

**Comment by richbrennan** [comment]
> @richbrennan I see this got marked as draft, rather than being marked for rereview - was that supposed to happen?

Yes, apologies for the confusion, I'd like to get the FE issues sorted before getting it re-reviewed.

File: source/sass/mozfest.scss
Comment (danielfmiranda) [inline source/sass/mozfest.scss:184]: Hi @richbrennan! Unfortunately, it looks like the tito widget is adding its own CSS that we cannot override to push its position lower unless we remove the styles completely as shown here: https://ti.to/docs/api/widget#tito-widget-v2-form-styles

Instead, I reduced the z-index of the nav bar so the widget can be completely shown instead of covered by the nav bar. Please let me know if this works, thanks!

Comment (richbrennan) [reply source/sass/mozfest.scss:184]: This works for me, I'll hand over to Marc to try, thanks Daniel!

**Review by Pomax — APPROVED** [review]

**Pull request merged**
